nemo_skills/inference/eval/bfcl.py

```diff
@@ -227,6 +228,15 @@ async def _generate_single_assistant_turn(self, inference_state_dict):
 
         input_dict = self.message_parser.construct_input_dict(messages, tools)
 
+        return_dict = {}
+        if self.cfg.count_prompt_tokens:
+            try:
+                num_input_tokens = get_token_count(self.hf_tokenizer, input_dict["prompt"])
+                return_dict["num_input_tokens"] = num_input_tokens
+            except ValueError as e:
```

Comment:
why do we need this exception? We don't have it in base class, so probably shouldn't have it here?

Reply:
Why can't we count tokens for messages? As in - the current code needs an update to do that, or it's not possible in principle?

Reply:
I don't think there's an easy way to do it tbh, at least from my cursory search I didn't find any solution. One contrived solution is to get the count from the ChatCompletion response, and somehow patch it with the user message lengths when calculating this for multi-turn.

Reply:
I don't fully understand - the problem is that we can't get counts for tool responses? As in why server side vs client side is a problem? The model will decide to call a tool and the output will be parsed by the server, not client, but I don't fully get why it affects input token counts. All the user messages should be available to us, should they not? Can you give an example of where this is tricky?